geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/EncryptorImpl.java

```diff
@@ -0,0 +1,769 @@
+          certificateMap.put(subject, cert);
+        }
+      }
+    }
+  }
+
+  /**
+   * Load the private key of the server. This method is not thread safe.
+   */
+  public static void initPrivateKey(Properties props) throws Exception {
+
+    String privateKeyFilePath = props.getProperty(PRIVATE_KEY_FILE_PROP);
+    privateKeyAlias = "";
+    privateKeyEncrypt = null;
+    if (privateKeyFilePath != null && privateKeyFilePath.length() > 0) {
+      KeyStore ks = KeyStore.getInstance("PKCS12");
+      privateKeyAlias = props.getProperty(PRIVATE_KEY_ALIAS_PROP);
+      if (privateKeyAlias == null) {
+        privateKeyAlias = "";
+      }
+      String keyStorePass = props.getProperty(PRIVATE_KEY_PASSWD_PROP);
+      char[] passPhrase = (keyStorePass != null ? keyStorePass.toCharArray() : null);
+      FileInputStream privateKeyFile = new FileInputStream(privateKeyFilePath);
+      try {
+        ks.load(privateKeyFile, passPhrase);
+      } finally {
+        privateKeyFile.close();
+      }
+      Key key = ks.getKey(privateKeyAlias, passPhrase);
+      Certificate keyCert = ks.getCertificate(privateKeyAlias);
+      if (key instanceof PrivateKey && keyCert instanceof X509Certificate) {
+        privateKeyEncrypt = (PrivateKey) key;
+        privateKeySignAlgo = ((X509Certificate) keyCert).getSigAlgName();
+        privateKeySubject = ((X509Certificate) keyCert).getSubjectDN().getName();
+      }
+    }
+  }
+
+  /**
+   * Initialize the Diffie-Hellman keys. This method is not thread safe
+   */
+  public static void initDHKeys(DistributionConfig config) throws Exception {
+
+    dhSKAlgo = config.getSecurityClientDHAlgo();
+    dhPrivateKey = null;
+    dhPublicKey = null;
+    // Initialize the keys when either the host is a client that has
+    // non-blank setting for DH symmetric algo, or this is a server
+    // that has authenticator defined.
+    if ((dhSKAlgo != null
+        && dhSKAlgo.length() > 0) /* || securityService.isClientSecurityRequired() */) {
+      KeyPairGenerator keyGen = KeyPairGenerator.getInstance("DH");
+      DHParameterSpec dhSpec = new DHParameterSpec(dhP, dhG, dhL);
+      keyGen.initialize(dhSpec);
+      KeyPair keypair = keyGen.generateKeyPair();
+
+      // Get the generated public and private keys
+      dhPrivateKey = keypair.getPrivate();
+      dhPublicKey = keypair.getPublic();
+
+      random = new SecureRandom();
+      // Force the random generator to seed itself.
+      byte[] someBytes = new byte[48];
+      random.nextBytes(someBytes);
+    }
+  }
+
+  public byte[] decryptBytes(byte[] data) throws Exception {
+    if (this.appSecureMode == CREDENTIALS_DHENCRYPT) {
+      String algo = null;
+      if (this.clientSKAlgo != null) {
+        algo = this.clientSKAlgo;
+      } else {
+        algo = dhSKAlgo;
+      }
+      Cipher c = getDecryptCipher(algo, this.clientPublicKey);
+      return decryptBytes(data, c);
+    } else {
+      return data;
+    }
+  }
+
+
```

Comment:
Looks like there are several places in this file that have a few extra newlines.